nordicsemi/dfu/dfu_transport_serial.py

```diff
@@ -306,7 +308,15 @@ def try_to_recover():
 
     def __ensure_bootloader(self):
         lister = DeviceLister()
-        device = lister.get_device(com=self.com_port)
+        device = None
+
+        start = datetime.now()
+        while datetime.now() - start < timedelta(seconds=self.DEFAULT_SERIAL_PORT_CONNECT_TIMEOUT):
+            device = lister.get_device(com=self.com_port)
+            if device:
+                break
+            time.sleep(0.05)
```

Comment:
Could make the busy loop less intensive by increasing sleep time to e.g. 0.5 seconds?